### PR TITLE
Mod 5

### PR DESCRIPTION
**Pull request by dawnmiceli100**
Added the following functionality:
1) User can re-order their queue
2) User can add and change ratings of videos in their queue

### CODE REVIEW

File: app/models/review.rb [hunk 1]
```diff
@@ -2,5 +2,7 @@ class Review < ActiveRecord::Base
   belongs_to :user
   belongs_to :video
 
-  validates_presence_of :rating, :body
+  #validates_presence_of :rating, :body
```
Comment (matthewbarram) [inline app/models/review.rb]: remove if not using

File: app/models/queue_item.rb [hunk 2]
```diff
@@ -3,15 +3,30 @@ class QueueItem < ActiveRecord::Base
...
+  validates_uniqueness_of :queue_position, scope: :user_id
+
+  validates_numericality_of :queue_position, { only_integer: true } 
 
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
```
Comment (matthewbarram) [inline app/models/queue_item.rb]: what if there isn't a review?

Comment (matthewbarram) [inline app/models/queue_item.rb]: Maybe you should return an empty string

File: app/controllers/queue_items_controller.rb [hunk 3]
```diff
@@ -41,15 +52,15 @@ def add_video_to_queue(video)
...
-    @current_user.queue_items.each do |item|
-      item.queue_position = new_position
-      item.save
-      new_position += 1
-    end
+  def update_queue_items
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |item|
+        queue_item = QueueItem.find(item["id"])
+        queue_item.update_attributes!(queue_position: item["queue_position"], rating: item["rating"]) if queue_item.user == current_user
```
Comment (matthewbarram) [inline app/controllers/queue_items_controller.rb]: This line lines is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits

File: app/models/queue_item.rb (hunk 2 above)
Comment (dawnmiceli100) [inline app/models/queue_item.rb]: Hi Matthew.
Thanks for reviewing my code. Would you please explain this to me? Is it better to return an empty string than nil? 
Thanks!Dawn

File: app/controllers/queue_items_controller.rb (hunk 3 above)
Comment (dawnmiceli100) [inline app/controllers/queue_items_controller.rb:62]: Thanks! I took care of this.

**Pull request merged**